### DISCUSSION TITLE
Packaging salt-lint for Fedora

### DISCUSSION THREAD

**Discussion by c4t3l** [warpnet/salt-lint#267] [General]
Greetings,

I hope this is an appropriate place to have this discussion... I wasn't sure if this should be in issues or not....

My name is Robby Callicotte and I am big fan of all things Salt Stack.  I wanted to reach out and let you know that I am in the process of building salt-lint for Fedora Linux.  The Bugzilla review request is located at https://bugzilla.redhat.com/show_bug.cgi?id=2013796.  

I hope we can work together to release this wonderful piece of software on Fedora.

**Comment by roaldnefs** [comment]
Thanks @c4t3l for you work on packaging salt-lint for Fedora. Let's hope the review process goes well. Please let us know if we can assist you in any way!

**Comment by jbouter** [comment]
Hi @c4t3l! 👋🏻 

We're really excited by your message :-) I'll keep an eye on the review process as well, and would like to thank you for your efforts thus far. Like @roaldnefs has said, please let us know if we can assist you!

**Comment by roaldnefs** [comment]
The required man page was added in #270 and just released in salt-lint version [0.7.0](https://pypi.org/project/salt-lint/).

**Comment by c4t3l** [comment]
Hello all.  I just wanted to let you know that salt-lint is now pushed to stable rpm repos for Fedora 34, 35, and 36 (rawhide).  In addition, the version for epel8 is currently in testing and should be promoted to the stable repos in 5 days.  

Thanks to @roaldnefs and @jbouter for all of your help!

**Reply by roaldnefs** [reply]
Thanks for your effort @c4t3l!

**Reply by jbouter** [reply]
Hi @c4t3l! I don't feel like I've assisted much, and most credit goes to yourself! I'd like to thank you for your efforts to make `salt-lint` available on Fedora!

What does the update process look like? Is there something watching our github releases page, would you prefer we notify you on release or could we perhaps create a CI trigger?

**Reply by c4t3l** [reply]
@jbouter, I can't believe I haven't seen your reply.  Sorry for taking so long to get back to you.  Fedora provides a release monitoring system called "Anitya".[1]  This system performs daily checks for updates (new tags/releases) in the upstream project repos.  This system will open a bugzilla ticket to notify the package maintainer of new updates.   In addition, it will kick off "scratch builds" to ensure that the existing build specfile can work with the update.


[1] - https://release-monitoring.org/project/240111/

**Comment by roaldnefs** [comment]
Hi @c4t3l, we've just released `v0.9.0`. Would it be possible to update the RPM package as well? Please let us know if you need our help in any way to support this.

**Reply by c4t3l** [reply]
Hello!  

I will start working on the packaging this weekend.  I will notify you if I need any assistance.  Thanks!

**Reply by c4t3l** [reply]
Just submitted updates to v0.9.1 to Fedora.  This includes the current releases (rawhide, f37, f36, epel9, epel8).  The rawhide release is available for immediate install.  The other releases will appear in official repos after 7 days or as soon as +3 karma points are accumulated. [1][2]


[1] - https://docs.fedoraproject.org/en-US/fesco/Updates_Policy/
[2] - https://bugzilla.redhat.com/show_bug.cgi?id=2160859

**Reply by jbouter** [reply]
@c4t3l Super! Thank you :)
